lib/hmmlearn/hmm.py

```diff
@@ -307,18 +309,246 @@ def _do_mstep(self, stats):
+            if self.n_trials < X.max():
+                raise ValueError(
+                    " Largest number of trials in a sample is {} " 
+                    " but the model takes max of {} trials".format(X.max(), self.n_trials))
+        else:
+            self.n_features = X.max() + 1
+
+            
+    def _init(self, X):
+        self._check_and_set_multinomial_n_features(X)
+        super()._init(X)
+        self.random_state = check_random_state(self.random_state)
+        
+        if isinstance(self.transmat_prior, np.ndarray):
+            self.transmat_ = self.transmat_prior.copy()
+        else:
+            self.transmat_ = np.random.random((self.n_components, self.n_components))
+            normalize(self.transmat_, axis=1)
+        
+        if isinstance(self.startprob_prior, np.ndarray):
+            self.startprob_ = self.startprob_prior.copy()
+        else:
+            self.startprob_ = np.random.random(self.n_components)
+            normalize(self.startprob_)
+
+        if isinstance(self.emission_prior, np.ndarray):
+            self.emissionprob_ = self.emission_prior.copy()
+        else:
+            self.emissionprob_ = np.random.random((self.n_components, self.n_features)) 
+            normalize(self.emissionprob_, axis=1)
+
+    def _check(self):
+        super()._check()
+
+        self.emissionprob_ = np.atleast_2d(self.emissionprob_)
+        n_features = getattr(self, "n_features", self.emissionprob_.shape[1])
+        if self.emissionprob_.shape != (self.n_components, n_features):
+            raise ValueError(
+                "emissionprob_ must have shape (n_components, n_features)")
+        else:
+            self.n_features = n_features
+
+    def _compute_likelihood(self, X):
+        probs = []
+        for component in range(self.n_components):
+            score = multinomial.pmf(X, n=self.n_trials, 
+                    p=self.emissionprob_[component,:])
+            probs.append(score)
+        return np.vstack(probs).T
+
+    def _compute_log_likelihood(self, X):
+        # X: n_samples x n_features
+        # return: n_samples x n_components 
+        logprobs = []
+        for component in range(self.n_components):
+            score = multinomial.logpmf(X, n=self.n_trials, 
+                    p=self.emissionprob_[component,:])
+            logprobs.append(score)
+        return np.vstack(logprobs).T
+
+    def _generate_sample_from_state(self, state, random_state=None):
+        random_state = check_random_state(random_state)
+        sample = multinomial.rvs(n=self.n_trials, 
+                p=self.emissionprob_[state,:],
+                size=1, random_state=self.random_state)
+        # sample is of shape (1, n_features); squeeze to keep dimensions consistent
+        return sample.squeeze(0)
+
+    def _initialize_sufficient_statistics(self):
+        stats = super()._initialize_sufficient_statistics()
+        stats['post'] = np.zeros(self.n_components)
```

Comment:
I guess you don't actually need to accumulate `post`, as `obs` has all the relevant info to normalize itself anyways?

Reply:
You're right; I took this out and re-tested.